cmd.go

```diff
@@ -278,8 +290,40 @@ func (i *Info) Help(f *gnuflag.FlagSet) []byte {
 	if i.Purpose != "" {
 		fmt.Fprintf(buf, "\nSummary:\n%s\n", strings.TrimSpace(i.Purpose))
 	}
+	hasSuperFlags := false
+	if superF != nil && len(i.ShowSuperFlags) != 0 {
+		filteredSuperF := gnuflag.NewFlagSetWithFlagKnownAs("", gnuflag.ContinueOnError, superF.FlagKnownAs)
+		contains := func(one string) bool {
+			for _, a := range i.ShowSuperFlags {
+				if strings.ToLower(one) == strings.ToLower(a) {
+					return true
+				}
+			}
+			return false
+		}
+		emptyFlag := &gnuflag.Flag{}
+		superF.VisitAll(func(flag *gnuflag.Flag) {
+			if contains(flag.Name) {
+				if found := filteredSuperF.Lookup(flag.Name); found == nil || found == emptyFlag {
```

Comment:
Can a flagset have multiple flags with the same name? From reading VisitAll it doesn't seem like it - it iterates through FlagSet.formal which is a map of the names.